src/audio/smart_amp/smart_amp.c

```diff
@@ -650,8 +650,14 @@ static int smart_amp_copy(struct comp_dev *dev)
 
 static int smart_amp_reset(struct comp_dev *dev)
 {
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+
 	comp_dbg(dev, "smart_amp_reset()");
 
+	sad->process = NULL;
```

Comment:
do we need to free this ?

Reply:
This points to a static function [`smart_amp_process`](https://github.com/thesofproject/sof/blob/main/src/audio/smart_amp/smart_amp.c#L532). It doesn't need to be freed.